amount_test.go

```diff
@@ -807,3 +809,40 @@ func TestAmount_Scan(t *testing.T) {
 		})
 	}
 }
+
+func TestAmount_RoundTo_is_concurrency_safe(t *testing.T) {
+	const n = 20
+
+	roundingModes := []currency.RoundingMode{
+		currency.RoundHalfUp,
+		currency.RoundHalfDown,
+		currency.RoundUp,
+		currency.RoundDown,
+	}
+
+	for i := range roundingModes {
+		roundingMode := roundingModes[i]
+
+		t.Run(fmt.Sprintf("rounding_mode_%d", roundingMode), func(t *testing.T) {
+			t.Parallel()
+
+			var allReadyWg, allDone sync.WaitGroup
+			allReadyWg.Add(n)
+			allDone.Add(n)
+
+			for i := 0; i < n; i++ {
+				go func() {
+					defer allDone.Done()
+					amount, _ := currency.NewAmount("10.99", "EUR")
+
+					allReadyWg.Done()
```

Comment:
I don't really understand what we're doing with allReadyWg, why is Done() followed by Wait()? Why would we wait inside an individual goroutine? Perhaps a code comment would help.

Reply:
Random passerby here: @bojanz is correct.

I checked, and the race detector still successfully flags this race if you delete `allReadyWg` entirely. (And if you change `n` above from `20` to `2`.) The race detector detects unsynchronized access, or that a data race _could have occurred given different timing_; it doesn't require that the timing for a data race actually occur.

Reply:
@bojanz My goal was to make the chances of a race condition higher by making sure that all goroutines will call the method virtually at the same moment, hence the waiting before the actual call in all of them. 

@josharian I have checked as well, you're right. I was always convinced that the race has to really occur, also in timing aspect. Something to learn every day. Thanks! :) 

I will simplify the test by reducing the goroutines amount to 2 and dropping `allReadyWg` completely.